### PR TITLE
Add support for Bitwuzla and CVC5

### PR DESCRIPTION
**Pull request by gussmith23**
Adds support for Bitwuzla and cvc5. 

I copied and pasted the files for cvc4 and boolector respectively to make the files for cvc5 and bitwuzla. They seem to be working! Thanks for making it so easy to add support. (And thanks @sorawee for the help on this!)

Note: I've only enabled QF_BV for both solvers, as this is all I need at the moment. If you want to enable new solvers, you can just add new entries to the `solver-features` list in each file. This will likely involve having to fix tests as well. 

- [x] Add tests (?)
- [x] Remove extra stuff from bitwuzla file (need help determining what's not needed)
- [ ] Remove extra stuff from cvc5 file (need help determining what's not needed)

### CODE REVIEW

File: rosette/solver/smt/bitwuzla.rkt [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+         (only-in "../../base/core/bitvector.rkt" bitvector bitvector? bv? bv bv-value @extract @sign-extend @zero-extend @bveq)
+         (only-in "../../base/core/function.rkt" function-domain function-range function? function fv)
+         (only-in "../../base/core/type.rkt" type-of)
+         (only-in "../../base/form/control.rkt" @if)
+         (only-in "../../base/core/reporter.rkt" current-reporter))
+
+(provide (rename-out [make-bitwuzla bitwuzla]) bitwuzla? bitwuzla-available?)
+
+(define-runtime-path bitwuzla-path (build-path ".." ".." ".." "bin" "bitwuzla"))
+(define bitwuzla-opts '("-m" "-i"))
```
Comment (sorawee) [inline rosette/solver/smt/bitwuzla.rkt:19]: I'm pretty sure `bitwuzla` doesn't support `-i`. Does it work for you?

Comment (gussmith23) [reply rosette/solver/smt/bitwuzla.rkt:19]: Weirdly, yes...

Comment (gussmith23) [reply rosette/solver/smt/bitwuzla.rkt]: Flag is removed!

**Comment by sorawee** [comment]
Adding tests would be nice. We have [GitHub Actions's workflow](https://github.com/emina/rosette/blob/master/.github/workflows/tests.yml) that downloads solvers and [tests](https://github.com/emina/rosette/blob/master/test/all-rosette-tests.rkt#L77) that invoke these solvers.

**Comment by gussmith23** [comment]
@sorawee added them, though I'm sure CI will break til I get all the dependencies right. Can you run the workflow?

**Comment by sorawee** [comment]
That would require @emina, @jamesbornholt, or @lukenels. I do not have a permission.

**Comment by gussmith23** [comment]
Bump @emina @jamesbornholt @lukenels -- could someone run the workflows?

**Comment by gussmith23** [comment]
Alternatively, deputize @sorawee so he can run them :)

Thanks all!!!

**Comment by sorawee** [comment]
The workflow now runs. It looks like you need a `sudo` for `apt-get install`.

**Comment by gussmith23** [comment]
@sorawee updated -- can you run again? or whoever ran it last time?

**Comment by gussmith23** [comment]
Ran into some weird behavior on boolector with the tests, see the above linked issue on Bitwuzla.

**Comment by gussmith23** [comment]
CI is passing. @sorawee @jamesbornholt @emina @lukenels ready for final review.

File: rosette/solver/smt/bitwuzla.rkt [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  (define raw-model (base/parse-solution server))
+  ; First, we need to fix up the model's shadowing of incremental variables
+  (define stripped-raw-model (if (hash? raw-model) (fixup-incremental-names raw-model) raw-model))
+  ; Now decode in an environment with fake types for UFs
+  (define m (decode stripped-raw-model (fake-env-types env)))
+  ; Finally, fix up the decoded model with the right types
+  (fixup-model m))
+
+
+; bitwuzla adds a prefix to constant names in incremental mode,
```
[the remark this answers is not in the document]
Comment (gussmith23) [reply rosette/solver/smt/bitwuzla.rkt]: Done! Can you run CI?

**Comment by sorawee** [comment]
This looks good to me, thanks!

**Comment by gussmith23** [comment]
Woohoo! Thank you!

**Pull request merged**